lib/src/net.rs

```diff
@@ -1,21 +1,20 @@
 use rand::Rng;
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use sha1::{Digest, Sha1};
 
-#[derive(Debug, Deserialize)]
+// see <https://developer.spotify.com/documentation/commercial-hardware/implementation/guides/zeroconf>
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct DeviceInfo {
     #[serde(rename = "deviceID")]
     pub device_id: String,
-    #[serde(rename = "remoteName")]
     pub remote_name: String,
-    #[serde(rename = "publicKey")]
     pub public_key: String,
-    #[serde(rename = "tokenType")]
-    pub token_type: Option<String>,
-    #[serde(rename = "clientID")]
-    pub client_id: Option<String>,
-    pub scope: Option<String>,
+    pub active_user: Option<String>, // undocumented but useful and returned by both librespot and librespot-java
```

Comment:
I tried both with an old 0.2.0 version and the latest 0.4.2 version of `librespot`. In each case, I got `Some("")` for the `activeUser` field. Does it work better on `librespot-java`?

Reply:
That is the expected behavior if the field is there but no user is currently connected. I did have to work around that for my use case but also it would be good to be able to tell if the field is missing/unsupported or if there is just no user connected right now. Ideas?

Reply:
I should have been more explicit. Even when connected and streaming music from `librespot`, the returned `activeUser` is empty. Which seems not the desired behavior. It's harmless, so I'll keep it for now, especially if it works well for your use case with `librespot-java`.

Concerning the distinction between the missing field and the empty field, I am not sure it is currently possible while using the `serde` to deserialize the `DeviceInfo` struct. It could be distinguished if the struct were deserialized manually, but that's kind of verbose, error prone and not very idiomatic.

Reply:
Oh, that's odd. It is definitely supposed to support it -- see librespot-org/librespot#1235 for the beginning of a fix.

Reply:
Really nice follow up! Thanks for taking so much time improving code, whatever the repository.

I tested your changes on `librespot`. It works perfectly and integrates nicely with the `spotify-connect` command. Well done!